File: go/p5/p5_v1.go
package main

import (
	"fmt"
	"math"
	"time"
)

func problem5() int {
	const to int = 20

	factors := make([]int, to)

	for i := 1; i <= to; i++ {
		counter := 2
		currentFactors := make([]int, i)

		num := i

		for num != 1 {
			if num%counter == 0 {
				num /= counter
				currentFactors[counter-1] += 1
			} else {
				counter += 1 + counter%2
			}

		}

		for j := range currentFactors {
			if currentFactors[j] > factors[j] {
				factors[j] = currentFactors[j]
			}
		}
	}

	var result int = 1
	
	for i, factor := range factors {
		result *= int(math.Pow(float64(i+1), float64(factor)))
	}

	return result
}

func main() {
	start := time.Now()
	fmt.Println("Result:", problem5())
	elapsed := time.Since(start)
	fmt.Printf("Execution time: %.6f seconds\n", elapsed.Seconds())
}

File: python/p5/p5_v1.py
def problem5():
    from functools import reduce

    to = 20

    factors = [0 for _ in range(to)]

    for number in range(1, to + 1):
        counter = 2
        current_factors = [0 for _ in range(number)]
        while number != 1:
            if number % counter == 0:
                number /= counter
                current_factors[counter - 1] += 1
            else:
                counter = counter + 1 if counter == 2 else counter + 2
        for number in range(len(current_factors)):
            if current_factors[number] > factors[number]:
                factors[number] = current_factors[number]

    return reduce(lambda x, y: x * y, [i ** factors[i - 1] for i in range(1, 20 + 1)])

if __name__ == "__main__":
    import time
    
    start_time = time.time()
    result = problem5()
    end_time = time.time()

    print(f"Result: {result}")
    print(f"Execution time: {end_time - start_time:.6f} seconds")